packages/spotlight/src/vite-plugin.ts

```diff
@@ -0,0 +1,260 @@
+    // eslint-disable-next-line no-param-reassign
+    colno = lineLength;
+  }
+
+  let start = Math.max(colno - 60, 0);
+  if (start < 5) {
+    start = 0;
+  }
+
+  let end = Math.min(start + 140, lineLength);
+  if (end > lineLength - 5) {
+    end = lineLength;
+  }
+  if (end === lineLength) {
+    start = Math.max(end - 140, 0);
+  }
+
+  newLine = newLine.slice(start, end);
+  if (start > 0) {
+    newLine = `'{snip} ${newLine}`;
+  }
+  if (end < lineLength) {
+    newLine += ' {snip}';
+  }
+
+  return newLine;
+}
+
+function isValidSentryStackFrame(frame: SentryStackFrame): frame is ValidSentryStackFrame {
+  return !!frame.filename && !!frame.lineno && !!frame.colno;
+}
+
+export const CONTEXT_LINES_ENDPOINT = '/spotlight/contextlines';
+export const sourceContextMiddleware: Connect.NextHandleFunction = function (req, res, next) {
+  // We're only interested in handling a PUT request to CONTEXT_LINES_ENDPOINT
+  if (req.url !== CONTEXT_LINES_ENDPOINT || req.method !== 'PUT') {
+    return next();
+  }
+
+  let requestBody = '';
+  req.on('data', chunk => {
+    requestBody += chunk;
+  });
+
+  req.on('end', async () => {
+    const stacktrace = parseStackTrace(requestBody);
+
+    if (!stacktrace) {
+      res.writeHead(500);
+      res.end();
+      return;
+    }
+
+    for (const frame of stacktrace.frames ?? []) {
+      if (
+        isValidSentryStackFrame(frame) &&
+        // let's ignore dependencies for now with this naive check
+        !frame.filename.includes('/node_modules/')
+      ) {
+        const generatedCode = await getGeneratedCodeFromServer(frame.filename);
+        if (!generatedCode) {
+          continue;
+        }
+
+        // Extract the inline source map from the minified code
+        const inlineSourceMapMatch = generatedCode.match(/\/\/# sourceMappingURL=data:application\/json;base64,(.*)/);
+
+        if (inlineSourceMapMatch && inlineSourceMapMatch[1]) {
+          const sourceMapBase64 = inlineSourceMapMatch[1];
+          const sourceMapContent = Buffer.from(sourceMapBase64, 'base64').toString('utf-8');
+          await applySourceContextToFrame(sourceMapContent, frame);
+        }
+      }
+    }
+
+    const responseJson = JSON.stringify(stacktrace);
+
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+    res.end(responseJson);
+  });
+};
+
+export default function spotlight({ port }: { port?: number } = {}): PluginOption {
+  Sentry.init({
+    tracesSampleRate: 0,
+    spotlight: true,
+  });
+
+  let spotlightPath: string;
+
+  return {
+    name: 'spotlight',
+    apply: 'serve',
+    transform(code: string, id: string /*, _opts = {}*/) {
+      // Only modify Vite's special client script
+      // The magic value below comes from vite/constants:CLIENT_ENTRY
+      if (!id.endsWith('vite/dist/client/client.mjs')) return;
+
+      const initSnippet = [
+        'createErrorOverlay=()=>{}',
+        `import * as Spotlight from '/@fs${spotlightPath}'`,
```

Comment:
are we missing a '/' here? Just asking because in `snippets.ts` we have the additional slash

```suggestion
        `import * as Spotlight from '/@fs/${spotlightPath}'`,
```

Reply:
Ah, good catch! I used to use `path.resolve()` when getting `spotlightPath` and then refactored the code a bit but forgot this part (and obviously didn't test this properly).

Reply:
Scratch that, we get a full URL and that's why it works.